Code/Legacy/CryCommon/CrySystemLoader.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <CrySystemLoader.h>
+
+#include <AzCore/Module/DynamicModuleHandle.h>
+#include <CryCommon/ISystem.h>
+
+constexpr const char* CrySystemLibraryName = AZ_TRAIT_OS_DYNAMIC_LIBRARY_PREFIX  "CrySystem" AZ_TRAIT_OS_DYNAMIC_LIBRARY_EXTENSION;
+
+constexpr const char* CreateInterfaceFunctionName = "CreateSystemInterface";
+
+constexpr const char* InjectEnvironmentFunctionName = "InjectEnvironment";
+constexpr const char* DetachEnvironmentFunctionName = "DetachEnvironment";
```

Comment:
We could rename the 
```
extern "C" AZ_DLL_EXPORT void InjectEnvironment(void* env);
extern "C" AZ_DLL_EXPORT void DetachEnvironment();
```
functions in [ISystem.h](https://github.com/o3de/o3de/blob/development/Code/Legacy/CryCommon/ISystem.h#L1107-L1108) to `InitializeDynamicModule` and `UninitializeDynamicModule` respectively.
The defintions of those functions are in [platform_impl.cpp](https://github.com/o3de/o3de/blob/development/Code/Legacy/CryCommon/platform_impl.cpp#L105-L119).
So once it is renamed there it should work like any other module

That way the regular DynamicModuleHandle could just be used directly.

Reply:
Great call, I'll do that instead